trial/src/main/kotlin/com/hartwig/actin/trial/TrialIngestion.kt

```diff
@@ -68,8 +68,8 @@ class TrialIngestion(private val eligibilityFactory: EligibilityFactory) {
                 ).right()
             else UnmappableTrial(trialId = trialState.trialId, trialErrors, unmappableCohorts).left()
         }
-        return if (trialsAndUnmappableTrial.any { it.isLeft }) trialsAndUnmappableTrial.mapNotNull { it.leftOrNull() }
-            .left() else trialsAndUnmappableTrial.mapNotNull { it.getOrNull() }.right()
+        return if (trialsAndUnmappableTrials.any { it.isLeft }) trialsAndUnmappableTrials.mapNotNull { it.leftOrNull() }
+            .left() else trialsAndUnmappableTrials.mapNotNull { it.getOrNull() }.right()
```

Comment:
I think this could be:
```suggestion
        val (errors, trials) = trialsAndUnmappableTrials.partitionAndJoin()
        return if (errors.isNotEmpty()) errors.left() else trials.right()
```